### ISSUE TITLE
Lots of depreciation errors with latest update

### ISSUE THREAD

**Issue by splendidrob** [spicywebau/craft-embedded-assets#253]
### Bug Description

Just ran the latest update and noticed a lot of depreciation errors popping up:

![screenshot_1390](https://github.com/spicywebau/craft-embedded-assets/assets/25520763/404ef141-79c7-4d22-832d-f0044136e912)

### Embedded Assets version

2.0.0

### Craft CMS version

Craft Solo 4.8.6

**Comment by ttempleton** [comment]
Could you please post one of the stack traces?

**Comment by splendidrob** [comment]
```

36	Deprecation error: The images embedded asset property has been deprecated, due to being removed in Embed 4. Use image instead.
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/spicyweb/craft-embedded-assets/src/models/EmbeddedAsset.php:209
35	spicyweb\embeddedassets\models\EmbeddedAsset::__construct(["title" => "The Alps 4K - Scenic Relaxation Film With Calming Music", "description" => "The Alps are possibly the most magical region in the world! Enjo...", "url" => "https://www.youtube.com/watch?v=BTMjD7_evjE", "type" => "video", ...])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/spicyweb/craft-embedded-assets/src/Service.php:535
34	spicyweb\embeddedassets\Service::_getDimensions(["title" => "The Alps 4K - Scenic Relaxation Film With Calming Music", "description" => "The Alps are possibly the most magical region in the world! Enjo...", "url" => "https://www.youtube.com/watch?v=BTMjD7_evjE", "type" => "video", ...])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/spicyweb/craft-embedded-assets/src/Service.php:303
33	spicyweb\embeddedassets\Service::createEmbeddedAsset(["title" => "The Alps 4K - Scenic Relaxation Film With Calming Music", "description" => "The Alps are possibly the most magical region in the world! Enjo...", "url" => "https://www.youtube.com/watch?v=BTMjD7_evjE", "type" => "video", ...])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/spicyweb/craft-embedded-assets/src/Service.php:254
32	spicyweb\embeddedassets\Service::getEmbeddedAsset(craft\elements\Asset)
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/spicyweb/craft-embedded-assets/src/Variable.php:29
31	spicyweb\embeddedassets\Variable::get(craft\elements\Asset)
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Extension/CoreExtension.php:1635
30	twig_get_attribute(craft\web\twig\Environment, Twig\Source, spicyweb\embeddedassets\Variable, "get", ...)
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/craftcms/cms/src/helpers/Template.php:129
29	craft\helpers\Template::attribute(craft\web\twig\Environment, Twig\Source, spicyweb\embeddedassets\Variable, "get", ...)
Called from /Users/Rob/Sites/craft-base/build/craft/templates/_includes/cp.media-text.twig:28
28	__TwigTemplate_b3da91afbdb8e18fd4470e3fec93a5d7::doDisplay(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], [])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:394
27	__TwigTemplate_b3da91afbdb8e18fd4470e3fec93a5d7::displayWithErrorHandling(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], [])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:367
26	__TwigTemplate_b3da91afbdb8e18fd4470e3fec93a5d7::display(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...])
Called from /Users/Rob/Sites/craft-base/build/craft/templates/_includes/cp.page-content.twig:25
25	__TwigTemplate_80a04f400da56967898ad5af7ee0ec64::doDisplay(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], [])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:394
24	__TwigTemplate_80a04f400da56967898ad5af7ee0ec64::displayWithErrorHandling(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], [])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:367
23	__TwigTemplate_80a04f400da56967898ad5af7ee0ec64::display(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...])
Called from /Users/Rob/Sites/craft-base/build/craft/templates/_pages/default.twig:28
22	__TwigTemplate_7f8db50f9be079ff9334866c9b605e65::block_mainContent(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], ["mainContent" => [__TwigTemplate_7f8db50f9be079ff9334866c9b605e65, "block_mainContent"]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:171
21	__TwigTemplate_72216030be2c8181d19edbd35927df5c::displayBlock("mainContent", ["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], ["mainContent" => [__TwigTemplate_7f8db50f9be079ff9334866c9b605e65, "block_mainContent"]])
Called from /Users/Rob/Sites/craft-base/build/craft/templates/_layouts/main.twig:71
20	__TwigTemplate_72216030be2c8181d19edbd35927df5c::doDisplay(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], ["mainContent" => [__TwigTemplate_7f8db50f9be079ff9334866c9b605e65, "block_mainContent"]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:394
19	__TwigTemplate_72216030be2c8181d19edbd35927df5c::displayWithErrorHandling(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], ["mainContent" => [__TwigTemplate_7f8db50f9be079ff9334866c9b605e65, "block_mainContent"]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:367
18	__TwigTemplate_72216030be2c8181d19edbd35927df5c::display(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], ["mainContent" => [__TwigTemplate_7f8db50f9be079ff9334866c9b605e65, "block_mainContent"]])
Called from /Users/Rob/Sites/craft-base/build/craft/templates/_pages/default.twig:1
17	__TwigTemplate_7f8db50f9be079ff9334866c9b605e65::doDisplay(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], ["mainContent" => [__TwigTemplate_7f8db50f9be079ff9334866c9b605e65, "block_mainContent"]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:394
16	__TwigTemplate_7f8db50f9be079ff9334866c9b605e65::displayWithErrorHandling(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], ["mainContent" => [__TwigTemplate_7f8db50f9be079ff9334866c9b605e65, "block_mainContent"]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:367
15	__TwigTemplate_7f8db50f9be079ff9334866c9b605e65::display(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], [])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/TemplateWrapper.php:45
14	Twig\TemplateWrapper::display(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...])
Called from /Users/Rob/Sites/craft-base/build/craft/templates/_pages/entry.twig:2
13	__TwigTemplate_60828a1a86d759390196e131207ce858::doDisplay(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], [])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:394
12	__TwigTemplate_60828a1a86d759390196e131207ce858::displayWithErrorHandling(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry], "craft" => craft\web\twig\variables\CraftVariable, "currentSite" => craft\models\Site, ...], [])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:367
11	__TwigTemplate_60828a1a86d759390196e131207ce858::display(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Template.php:379
10	__TwigTemplate_60828a1a86d759390196e131207ce858::render(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/TemplateWrapper.php:38
9	Twig\TemplateWrapper::render(["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/twig/twig/src/Environment.php:280
8	craft\web\twig\Environment::render("_pages/entry.twig", ["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/craftcms/cms/src/web/View.php:482
7	craft\web\View::renderTemplate("_pages/entry.twig", ["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry]])
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/craftcms/cms/src/web/View.php:535
6	craft\web\View::renderPageTemplate("_pages/entry.twig", ["entry" => craft\elements\Entry, "variables" => ["entry" => craft\elements\Entry]], "site")
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/craftcms/cms/src/web/TemplateResponseFormatter.php:57
5	craft\web\TemplateResponseFormatter::format(craft\web\Response)
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/yiisoft/yii2/web/Response.php:1100
4	craft\web\Response::prepare()
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/craftcms/cms/src/web/Response.php:337
3	craft\web\Response::prepare()
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/yiisoft/yii2/web/Response.php:340
2	craft\web\Response::send()
Called from /Users/Rob/Sites/craft-base/build/craft/vendor/yiisoft/yii2/base/Application.php:390
1	craft\web\Application::run()
Called from /Users/Rob/Sites/craft-base/build/public_html/index.php:12
```

**Comment by ttempleton** [comment]
Thanks for posting that @splendidrob. [Those deprecation warnings are intentionally being logged for embedded assets with those properties](https://github.com/spicywebau/craft-embedded-assets/blob/4.0.0/src/models/EmbeddedAsset.php#L202-L239), so this isn't a bug. However, I do understand that the deprecation warnings might not be relevant for properties that aren't actually used in the Craft project, and it would be worth improving the deprecation logging to only happen when a property is actually accessed.

**Comment by ttempleton** [comment]
Just released 4.0.1 and 5.0.1 which limit the deprecation logging to front-end requests.

**Comment by splendidrob** [comment]
@ttempleton All makes sense! Thanks very much for the update...much appreciated!
